src/dbnode/server/server.go

```diff
@@ -183,6 +194,25 @@ func Run(runOpts RunOptions) {
 		logger.Fatalf("could not resolve local host ID: %v", err)
 	}
 
+	var (
+		tracer      opentracing.Tracer
+		traceCloser io.Closer
+	)
+
+	if cfg.Tracing == nil {
+		tracer = opentracing.NoopTracer{}
+		logger.Infof("tracing disabled for %s; set `tracing.backend` to enable", serviceName)
+	} else {
+		// setup tracer
+		tracer, traceCloser, err = cfg.Tracing.NewTracer(serviceName, scope, zapLogger)
+		if err != nil {
```

Comment:
Is there any way where we can keep the tracing configuration in config by default and have it just not break anything if jaeger is not available? I don't know how the buffering / the library work internally

Reply:
so iiuc, traces are sent over UDP and will just silently fail if there is no Jaeger collector setup. However, if you don't have the Jaeger config at all, a no-op tracer is created which doesn't send any traces.